packages/grid/data-grid/src/DataGrid.tsx

```diff
@@ -68,7 +68,7 @@ DataGrid2.propTypes = {
       );
     }
   }),
-  columns: chainPropTypes(PropTypes.any, (props) => {
+  columns: chainPropTypes(PropTypes.any, (props: DataGridProps) => {
     if (props.columns && props.columns.some((column) => column.resizable)) {
```

Comment:
We should likely have a ts-expect-error but I guess it's simpler to only cover this case with prop-types only.